witness/src/Dscp/DB/Rocks/Real/Functions.hs

```diff
@@ -46,28 +77,64 @@ closeNodeDB = closeRocksDB . _rdDatabase
 -- Reading/writing
 ------------------------------------------------------------
 
--- | Read ByteString from RocksDb using given key.
-rocksGetBytes :: MonadIO m => ByteString -> DB -> m (Maybe ByteString)
-rocksGetBytes key DB {..} = Rocks.get rocksDB rocksReadOpts key
+get :: (HasRocksDB ctx m, Serialise a, Serialise b) => a -> m (Maybe b)
+get a = do
+    db  <- getDB
+    mbs <- rocksGetBytes db (serialise' a)
```

Comment:
`rocksGetBytes db (serialise' a) >>= \case` is a preferred method of doing what you want (it explicitly states that `mbs` is used nowhere except from the match itself).